wtpython/__main__.py

```diff
@@ -2,16 +2,53 @@
 import runpy
 import sys
 import traceback
+import types
 
 import pyperclip
 from rich import print
+from rich.markdown import HorizontalRule
+from rich.traceback import Traceback
 
 from wtpython import SearchError
 from wtpython.backends.stackoverflow import StackOverflowFinder
 from wtpython.display import Display, store_results_in_module
 from wtpython.settings import GH_ISSUES, MAX_SO_RESULTS
 
 
+def trim_exception_traceback(tb: traceback) -> traceback:
+    """
+    Trim the traceback to remove extra frames
+
+    Because of the way we are currently running the code, any traceback
+    created during the execution of the application will be include the
+    stack frames of this application. This function removes all the stack
+    frames from the beginning of the traceback until we stop seeing `runpy`.
+    """
+    seen_runpy = False
+    collecting = False
+
+    frames = []
+    while tb is not None:
+        cur = tb.tb_frame
+        filename = cur.f_code.co_filename
+
+        if "runpy" in filename:
+            seen_runpy = True
+        elif seen_runpy:
+            collecting = True
```

Comment:
can we `break` here and just return `tb` ? I think that works .. with a slight modification (where you have to keep continuing till the first non-runpy codeline)

```py
seen_runpy = False
tb = exc.__traceback__
while tb is not None:
    cur = tb.tb_frame
    filename = cur.f_code.co_filename
    if "runpy" in filename:
        seen_runpy = True
    elif seen_runpy and "runpy" not in filename:
        break
    tb = tb.tb_next
exc.__traceback__ = tb
traceback.print_exception(type(exc), exc, exc.__traceback__)
```

Reply:
Ah, you're right. I was messing around with another example I found online generating a traceback from scratch, and I got too tunnel visioned to see this. Thanks!

Reply:
should the check be an equality with `runpy.py` instead? this can potentially throw off cases where the driver file is say `runpy_xyz.py`

Reply:
maybe changing `filename = cur.f_code.co_filename` to `filename = Path(cur.f_code.co_filename).name` where `Path` is `pathlib.Path`